### PR TITLE
feat: add precompiled DOCA instructions and notes

### PR DESCRIPTION
**Pull request by maze88**
Prepared documentation and notes for upcoming precompiled DOCA feature.

### CODE REVIEW

File: docs/platform-support.rst [hunk 1]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+
+--------
+Overview
+--------
+
+To save startup (compilation) time and operational effort, precompiled DOCA driver images are available for common OS/flavor/kernel variants.
+
+The container image tag pattern used for common variants is: **driver_ver-container_ver-kernel_ver-flavor-os**. For example: ``24.07-0.6.1.0-6.8.0-49-generic-ubuntu24.04``
+
+**NOTE:** The tags are multi-architecture manifests, so no need to state the architecture.
```
Comment (e0ne) [inline docs/platform-support.rst:231]: are you going to propose a corresponding change to network operator? precompiled image tag requires CPU architecture now [1]

[1] https://github.com/Mellanox/network-operator/blob/e44a4bd19135374afb5b0aab32e2af2c688751cc/pkg/state/state_ofed.go#L68

File: docs/platform-support.rst [hunk 2]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+
+**NOTE:** For the *generic* flavor of Ubuntu, the default Kernel version is used for precompiling (e.g. *6.8.0-31* for Ubuntu 24.04). Whereas for all other flavors, their latest (at time of DOCA packaging/release) Kernel version is used.
+
+----------------------------
+Limitations and Restrictions
+----------------------------
+
+- Support for deploying the driver containers with precompiled drivers is currently limited to hosts with Ubuntu 22.04 or 24.04 (both amd64 and arm64).
+- NVIDIA supports precompiled driver containers for the most recently released DOCA-OFED GA drivers.
+- NVIDIA builds precompiled driver containers for generic, nvidia, aws, azure, and oracle kernel variants.
```
Comment (e0ne) [inline docs/platform-support.rst:241]: 'nvidia, aws, azure, and oracle' => 'NVIDIA, AWS, Azure, and Oracle'

File: docs/platform-support.rst [hunk 3]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+- If your hosts use a different kernel variant, you can create a custom precompiled driver container and host it in your own container registry. Please refer to :ref:`advanced-configurations-precompiled` section, below.
+- Precompiled driver containers are currently unsigned.
+
+.. warning::
+   - Only generic kernel variant is tested and supported as a GA.
+   - nvidia, aws, azure and oracle kernel variants are supported as a Tech Preview and have limited testing.
```
Comment (e0ne) [inline docs/platform-support.rst:247]: the same as above

**Review by e0ne — CHANGES_REQUESTED** [review]

File: docs/platform-support.rst (hunk 1 above)
Comment (maze88) [reply docs/platform-support.rst:231]: Oh, I wasn't aware of that...

The simplest workaround would be for me to skip creating manifests, and just deliver both architecture tags explicitly (but that'd be a step backwards)...
Alternatively, how simple/quick would it be to adjust the `state_ofed.go` code to make the `-<cpu-arch>` suffix _optional_? (i.e. if it's not stated then to assume an appropriate multi-arch manifest is provided)

File: docs/platform-support.rst (hunk 2 above)
Comment (maze88) [reply docs/platform-support.rst:241]: and 'generic' should remain lowercase?

File: docs/platform-support.rst (hunk 1 above)
Comment (maze88) [reply docs/platform-support.rst]: Ok, the tags with explicit architecture are still pushed so this code/automation should work fine.

And for cases where a user may want to explicitly/manually set the DOCA version - then they may choose to use the manifest tag (which points to both architectures).

File: docs/platform-support.rst (hunk 2 above)
Comment (rollandf) [reply docs/platform-support.rst]: I suggest to have quotes around the variant names and keep the same as shown in the CLI.
`generic`, `nvidia`, `aws` ....

File: docs/platform-support.rst [hunk 4]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
      - Yes
      - 
 
+
+=============================================
+Supported Precompiled Images for DOCA Drivers
```
Comment (rollandf) [inline docs/platform-support.rst:220]: Supported Precompiled Container Images for DOCA Drivers

File: docs/platform-support.rst [hunk 5]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+=============================================
+
+--------
+Overview
+--------
+
+To save startup (compilation) time and operational effort, precompiled DOCA driver images are available for common OS/flavor/kernel/architecture variants.
```
Comment (rollandf) [inline docs/platform-support.rst:227]: Suggestion:
remove : (compilation)
add: precompiled DOCA driver **container** images

Comment (rollandf) [inline docs/platform-support.rst:227]: What is a flavor?

File: docs/platform-support.rst [hunk 6]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+
+The container image tag pattern used for common variants is: **driver_ver-container_ver-kernel_ver-flavor-os-arch**. For example: ``24.07-0.6.1.0-6.8.0-49-generic-ubuntu24.04-amd64``
```
Comment (rollandf) [inline docs/platform-support.rst:229]: Why is it needed?
What the user will do with that?

File: docs/platform-support.rst [hunk 7]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+
+**NOTE:** The images are also packaged as multi-architecture manifests, so manually stating architecture component is optional.
```
Comment (rollandf) [inline docs/platform-support.rst:231]: Not needed.
The only thing the user can do is specify the DOCA driver version in the NicClusterPolicy.
It cannot define kernel/arch or anything like that

File: docs/platform-support.rst [hunk 8]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+
+**NOTE:** For the *generic* flavor of Ubuntu, the default Kernel version is used for precompiling (e.g. *6.8.0-31* for Ubuntu 24.04). Whereas for all other flavors, their latest (at time of DOCA packaging/release) Kernel version is used.
+
+----------------------------
+Limitations and Restrictions
+----------------------------
+
+- Support for deploying the driver containers with precompiled drivers is currently limited to hosts with Ubuntu 22.04 or 24.04 (both amd64 and arm64).
```
Comment (rollandf) [inline docs/platform-support.rst:239]: Why make it a limitation/restriction?
Supported Precompiled DOCA Driver container images:
- Ubuntu 22.04 amd64/arm64
- Ubuntu 24.04  amd64/arm64

File: docs/platform-support.rst [hunk 9]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+- NVIDIA supports precompiled driver containers for the most recently released DOCA-OFED GA drivers.
```
Comment (rollandf) [inline docs/platform-support.rst:240]: Not sure what this means.
Each Network Operator version comes with a specific DOCA driver version.

Also `OFED` should not be used

File: docs/platform-support.rst [hunk 10]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+- NVIDIA builds precompiled driver containers for generic, NVIDIA, AWS, Azure, and Oracle kernel variants.
+- If your hosts use a different kernel variant, you can create a custom precompiled driver container and host it in your own container registry. Please refer to :ref:`advanced-configurations-precompiled` section, below.
+- Precompiled driver containers are currently unsigned.
```
Comment (rollandf) [inline docs/platform-support.rst:243]: move the unsigned part before the custom made part

File: docs/platform-support.rst [hunk 11]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+
+.. warning::
+   - Only generic kernel variant is tested and supported as a GA.
```
Comment (rollandf) [inline docs/platform-support.rst:246]: add quotes on kernel variant name: `generic`

File: docs/platform-support.rst [hunk 12]
```diff
@@ -215,3 +215,33 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+   - NVIDIA, AWS, Azure, and Oracle kernel variants are supported as a Tech Preview and have limited testing.
```
Comment (rollandf) [inline docs/platform-support.rst:247]: add quotes on kernel variant names: `nvidia` .....

**Review by rollandf — CHANGES_REQUESTED** [review]

File: docs/platform-support.rst (hunk 5 above)
Comment (maze88) [reply docs/platform-support.rst:227]: Kernel flavors: NVIDIA, AWS, Azure, and Oracle...

File: docs/platform-support.rst (hunk 9 above)
Comment (maze88) [reply docs/platform-support.rst:240]: I don't know about that. @kvadrage (Alex.P) wrote it...

How would you suggest to alter these notes then?

Comment (rollandf) [reply docs/platform-support.rst]: I suggest to remove this sentence.
Maybe add link to NGC in the section before.
See available precompiled container images in [NGC](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/mellanox/containers/doca-driver/tags).

I will let @kvadrage comment here.

**Review by rollandf — APPROVED** [review]
LGTM

File: docs/platform-support.rst [hunk 13]
```diff
@@ -215,3 +215,39 @@ NVIDIA Network Operator has been validated in the following scenarios:
...
+Supported Precompiled Container Images for DOCA Drivers
+=======================================================
+
+--------
+Overview
+--------
+
+To save startup time and operational effort, precompiled DOCA driver container images are available for common OS/flavor/kernel/architecture variants.
+
+The container image tag pattern used for common variants is: **driver_ver-container_ver-kernel_ver-flavor-os-arch**. For example: ``24.07-0.6.1.0-6.8.0-49-generic-ubuntu24.04-amd64``
```
Comment (e0ne) [inline docs/platform-support.rst:229]: thanks @rollandf  to pointing me that it's not a correct image format. please make it exactly the same as mentioned here: https://github.com/Mellanox/network-operator/blob/295f5f90eef5dc0db4681a28e348900322dfe724/pkg/state/state_ofed.go#L68

**Pull request merged**